src/SFA.DAS.RoatpAssessor/SFA.DAS.RoatpAssessor.Web/Controllers/RoatpAssessorControllerBase.cs

```diff
@@ -0,0 +1,116 @@
+        {
+            if (command?.Status == null) return string.Empty;
+            command.OptionInProgressText = command.Status == AssessorPageReviewStatus.InProgress && !string.IsNullOrEmpty(command.OptionInProgressText) ? command.OptionInProgressText : string.Empty;
+            command.OptionPassText = command.Status == AssessorPageReviewStatus.Pass && !string.IsNullOrEmpty(command.OptionPassText) ? command.OptionPassText : string.Empty;
+            command.OptionFailText = command.Status == AssessorPageReviewStatus.Fail && !string.IsNullOrEmpty(command.OptionFailText) ? command.OptionFailText : string.Empty;
+
+            switch (command.Status)
+            {
+                case AssessorPageReviewStatus.Pass:
+                    return command.OptionPassText;
+                case AssessorPageReviewStatus.Fail:
+                    return command.OptionFailText;
+                case AssessorPageReviewStatus.InProgress:
+                    return command.OptionInProgressText;
+                default:
+                    return string.Empty;
+            }
+        }
+
+        protected async Task<IActionResult> ValidateAndUpdatePageAnswer<T>(SubmitAssessorPageAnswerCommand command,
+                                                          Func<Task<T>> viewModelBuilder,
+                                                          string errorView) where T : ReviewAnswersViewModel
+        {
+            var validationResponse = await AssessorPageValidator.Validate(command);
+            if (validationResponse.Errors != null && validationResponse.Errors.Any())
+            {
+                var viewModel = await viewModelBuilder.Invoke();
+                viewModel.Status = command.Status;
+                viewModel.OptionFailText = command.OptionFailText;
+                viewModel.OptionInProgressText = command.OptionInProgressText;
+                viewModel.OptionPassText = command.OptionPassText;
+                viewModel.ErrorMessages = validationResponse.Errors;
+                return View(errorView, viewModel);
+            }
+            else
+            {
+                await SubmitAssessorPageOutcome(command);
+
+                if (string.IsNullOrEmpty(command.NextPageId))
+                {
+                    return RedirectToAction("ViewApplication", "Overview", new { applicationId = command.ApplicationId });
+                }
+                else
+                {
+                    return RedirectToAction("ReviewPageAnswers", new { applicationId = command.ApplicationId, sequenceNumber = command.SequenceNumber, sectionNumber = command.SectionNumber, pageId = command.NextPageId });
+                }
+            }
+        }
+
+        protected async Task SubmitAssessorPageOutcome(SubmitAssessorPageAnswerCommand command)
+        {
+            var userId = HttpContext.User.UserId();
+            userId = "temp"; //TODO: Can't access the user until staff idams is enabled
```

Comment:
same as previous 'temp' comment

Reply:
Awaiting iDAMS to work correctly. A ticket will be raised to sort this out.